File: src/Players/HumanPlayer.php
<?php

namespace App\Players;

use App\Card;

class HumanPlayer extends Player
{
    public function chooseAction(): void
    {
        $players = $this->showdown->getPlayers();

        // If there are more than one player and exchange permission then choose action
        if (count($players) > 1 && $this->exchangePermission) {
            // 選擇行動
            echo "{$this->getName()} 選擇行動: 1. 換牌 2. 出牌\n";
            $action = trim(fgets(STDIN));
            if ($action == 1) {
                // 換牌
                echo "{$this->getName()} 選擇換牌\n";
                //                $this->exchangeCard();
            } else {
                if ($action != 2) {
                    echo "輸入錯誤, 請重新輸入\n";
                    $this->chooseAction();

                    return;
                }
            }

            // 1. 換牌 2. 出牌
            // 選擇 2 跳出 if 執行後面的程式碼

        }

        echo "{$this->getName()} 的手牌: ";
        foreach ($this->getHand() as $index => $card) {
            /** @var Card $card */
            echo "[$index] {$card->getSuit()->toCardString()}{$card->getRank()->toCardString()} ";
        }
        echo "\n請選擇要出的牌: ";

        $target = trim(fgets(STDIN));

        if (empty($this->hand->cards[$target])) {
            echo "輸入錯誤, 請重新輸入\n";
            $this->chooseAction();
        } else {
            // Show card and
            $card = $this->hand->cards[$target];
            $this->showCard($card);

            // Remove card from hand
            unset($this->hand->cards[$target]);
            $this->hand->cards = array_values($this->hand->cards);
        }
    }

    public function nameHimself(): void
    {
        $name = fgets(STDIN);
        $name = trim($name);
        $this->setName($name);
    }
}

File: showdown-go/entity/player/human_player.go
package player

import (
	"fmt"
	"math/rand"
	"showdown-go/entity"
)

type HumanPlayer struct {
	AbstractPlayer
}

func NewHumanPlayer(showdown *entity.Showdown) *HumanPlayer {
	return &HumanPlayer{
		AbstractPlayer{
			Hand:     entity.NewHand(),
			Showdown: *showdown,
		},
	}
}

func (h *HumanPlayer) ChooseAction() {
	// Randomly choose action
	if rand.Intn(2) == 1 {
		// exchange card and can't exchange myself
		num := rand.Intn(len(h.Showdown.GetPlayers()))
		player := h.Showdown.GetPlayers()[num]

		for player.GetName() == h.GetName() {
			num := rand.Intn(len(h.Showdown.GetPlayers()))
			player = h.Showdown.GetPlayers()[num]
		}

		exchangeHand := entity.NewExchangeHands(h, player)
		h.Showdown.AddExchangeHands(exchangeHand)
		h.SetExchangePermission(false)
	}

	targetPlayer := h.CheckChangeHands()
	targetCard := rand.Intn(len(targetPlayer.GetHand()))
	card := targetPlayer.GetHand()[targetCard]
	h.ShowCard(card)

	// Remove card from hand
	targetPlayer.GetHand()[targetCard] = nil
	targetPlayer.SetHand(targetPlayer.GetHand())
}

func (h *HumanPlayer) NameHimself() {
	// in go
	var name string
	fmt.Println("請輸入名稱: ")
	if _, err := fmt.Scanln(&name); err != nil {
		fmt.Println(err)
	}

	h.SetName(name)
}
